browser/resources/settings/email_aliases_page/tests/email_aliases.test.tsx

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import { render, screen, fireEvent, waitFor, act } from '@testing-library/react'
+import { ManagePage } from '../email_aliases'
+import { MappingService } from '../content/types'
+import { localeRegex } from './test_utils'
+
+// Mock the mapping service
+const mockMappingService: MappingService = {
+  getAccountEmail: jest.fn(),
+  logout: jest.fn(),
+  requestAccount: jest.fn(),
+  getAliases: jest.fn(),
+  createAlias: jest.fn(),
+  updateAlias: jest.fn(),
+  deleteAlias: jest.fn(),
+  generateAlias: jest.fn(),
+  onAccountReady: jest.fn(),
+  cancelAccountRequest: jest.fn(),
+  fillField: jest.fn(),
+  showSettingsPage: jest.fn()
+}
+
+describe('ManagePage', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('shows loading state initially', () => {
+    render(<ManagePage mappingService={mockMappingService} />)
+    expect(document.querySelector('[name="loading-spinner"]'))
+      .toBeInTheDocument()
+    expect(screen.getByText(localeRegex(
+      'emailAliasesConnectingToBraveAccount'))).toBeInTheDocument()
+  })
+
+  it('shows sign up form when no email is available', async () => {
+    mockMappingService.getAccountEmail = jest.fn().mockResolvedValue(null)
+
+    await act(async () => {
+      render(<ManagePage mappingService={mockMappingService} />)
+    })
+
+    await waitFor(() => {
+      expect(screen.getByText(localeRegex('emailAliasesSignInOrCreateAccount')))
+        .toBeInTheDocument()
+      expect(screen.getByPlaceholderText(localeRegex(
+        'emailAliasesEmailAddressPlaceholder'))).toBeInTheDocument()
+    })
+  })
+
+  it('shows main view when email is available', async () => {
+    const mockEmail = 'test@brave.com'
+    const mockAliases = [
+      {
+        email: 'alias1@brave.com',
+        note: 'Test Alias 1',
+      }
+    ]
+    mockMappingService.getAccountEmail = jest.fn().mockResolvedValue(mockEmail)
+    mockMappingService.getAliases = jest.fn().mockResolvedValue(mockAliases)
+
+    await act(async () => {
+      render(<ManagePage mappingService={mockMappingService} />)
+    })
+
+    await waitFor(() => {
+      expect(screen.getByText(mockEmail)).toBeInTheDocument()
+      expect(screen.getByText(/alias1@brave\.com/)).toBeInTheDocument()
+      expect(screen.getByText(/Test Alias 1/)).toBeInTheDocument()
+    })
+  })
+
+  it('handles sign up flow', async () => {
```

Comment:
@fallaciousreasoning shouldn't we also be checking the state to make sure it's correct? Like `accountState`, etc...?

Reply:
I guess `accountState` is actually part of the service and not react so to clarify I mean react state

Reply:
Hey so I'd really prefer not to test the React state - Its an internal detail of the component (I'd argue similar to private members on a class).

The result of a component is basically
`(props, state) => DOM`
so I think what we care about is that the DOM tree looks right.

As an example:

```
function Input() {
  const [text, setText] = useState('')
  return <input value={text} onChange={e => setText(e.target.value)} />
}
```

if we were testing that, we don't care if the text is right inside the Input component state after sending some keystrokes, we care that the generated DOM has the correct value.